app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -1149,39 +1197,254 @@ export class FilterChangesList extends React.Component<
     const disableAllCheckbox =
       files.length === 0 || isCommitting || rebaseConflictState !== null
 
-    const toBeCommittedFilterText = this.state.filterToIncludedCommit
-      ? 'Show files included and not included in the commit'
-      : 'Only show files to be included in the commit'
+    const fileCount = files.length
+
+    const filesChecked = workingDirectory.files.filter(
+      f => f.selection.getSelectionType() !== DiffSelectionType.None
+    )
+
+    const showHiddenCheckedFilesWarning = this.isCommittingFileHiddenByFilter(
+      this.state.filterText,
+      filesChecked.map(f => f.id),
+      this.state.filteredItems,
+      fileCount
+    )
 
     return (
-      <div
-        className="header"
-        onContextMenu={this.onContextMenu}
-        ref={this.headerRef}
-      >
+      <div className="checkbox-container">
         <Checkbox
           ref={this.includeAllCheckBoxRef}
           value={includeAllValue}
           onChange={this.onIncludeAllChanged}
           disabled={disableAllCheckbox}
           ariaLabelledBy="changes-list-check-all-label"
+          className="changes-list-check-all"
         />
 
-        <Button
-          onClick={this.onFilterToIncludedInCommit}
-          className={classNames({
-            'included-in-commit-filter-on': this.state.filterToIncludedCommit,
-          })}
-          ariaLabel={toBeCommittedFilterText}
-          tooltip={toBeCommittedFilterText}
-        >
-          <Octicon symbol={octicons.filter} />
-        </Button>
-        <label id="changes-list-check-all-label"> {filesDescription}</label>
+        <label id="changes-list-check-all-label">{visibleFiles} files</label>
+        <span className="spacer"></span>
+        {showHiddenCheckedFilesWarning ? (
+          <span>
+            <Button
+              className="filter-button hidden-files-warning"
+              ariaExpanded={this.state.isFilterOptionsOpen}
+              ariaLabel="Open Files Hidden by Filter Warning"
+              tooltip="Open Files Hidden by Filter Warning"
+              onClick={this.openFilterWarning}
+              onButtonRef={this.onFilterWarningButtonRef}
+            >
+              <Octicon symbol={octicons.alert} />
+            </Button>
+            {this.state.isFilesWarningOpen && this.renderFilterWarning()}
+          </span>
+        ) : null}
+
+        {this.state.filterToIncludedCommit ? (
+          <Button
+            className="filter-button remove-filter"
+            onClick={this.clearNonTextFilters}
+            ariaLabel="Clear 1 additional filter(s)"
+            tooltip="Clear 1 additional filter(s)"
```

Comment:
WDYT about "Clear filter option(s)"?
```suggestion
            ariaLabel="Clear filter option(s)"
            tooltip="Clear filter option(s)"
```

It should be relatively clear given that the menu has the same title at least!